website/content/api-docs/admin-partitions.mdx

```diff
@@ -0,0 +1,258 @@
+---
+layout: api
+page_title: Admin Partition - HTTP API
+description: The /partition endpoints allow for managing Consul Enterprise Admin Partitions.
+---
+
+# Admin Partition - HTTP API
+
+<EnterpriseAlert />
+
+The functionality described here is available only in
+[Consul Enterprise](https://www.hashicorp.com/products/consul/) version 1.11.0 and later.
+
+## Create a Partition
+
+This endpoint creates a new Partition.
+
+| Method | Path         | Produces           |
+| ------ | ------------ | ------------------ |
+| `PUT`  | `/partition` | `application/json` |
+
+The table below shows this endpoint's support for
+[blocking queries](/api/features/blocking),
+[consistency modes](/api/features/consistency),
+[agent caching](/api/features/caching), and
+[required ACLs](/api#authentication).
+
+| Blocking Queries | Consistency Modes | Agent Caching | ACL Required     |
+| ---------------- | ----------------- | ------------- | ---------------- |
+| `NO`             | `none`            | `none`        | `operator:write` |
+
+### Parameters
+
+- `Name` `(string: <required>)` - The partition name. This must be a valid
+  DNS hostname label.
+
+- `Description` `(string: "")` - Free form partition description.
+
+### Sample Payload
+
+```json
+{
+  "Name": "na-west",
+  "Description": "Partition for North America West",
+}
+```
+
+### Sample Request
+
+```shell-session
+$ curl -X PUT \
+   -H "X-Consul-Token: 5cdcae6c-0cce-4210-86fe-5dff3b984a6e" \
+   --data @payload.json \
+   http://127.0.0.1:8500/v1/partition
+```
+
+### SampleResponse
+
+```json
+{
+  "Name": "na-west",
+  "Description": "Partition for North America West",
+  "CreateIndex": 55,
+  "ModifyIndex": 55
+}
+```
+
+## Read a Partition
+
+This endpoint reads a Partition with the given name.
+
+| Method | Path               | Produces           |
+| ------ | ------------------ | ------------------ |
+| `GET`  | `/partition/:name` | `application/json` |
+
+The table below shows this endpoint's support for
+[blocking queries](/api/features/blocking),
+[consistency modes](/api/features/consistency),
+[agent caching](/api/features/caching), and
+[required ACLs](/api#authentication).
+
+| Blocking Queries | Consistency Modes | Agent Caching | ACL Required                          |
+| ---------------- | ----------------- | ------------- | ------------------------------------- |
+| `NO`             | `consistent`      | `none`        | `operator:read` or `none`<sup>1</sup> |
+
+<sup>1</sup> A non-anonymous token in a non-default partition can read its own partition.
```

Comment:
Would like some eyes here

Reply:
For the consistency modes, stale queries are supported as well (so it should be `all`).

The token also doesn't have to be non-default, just non-anonymous.

Reply:
Are stale queries supported? I saw this TODO in the code: https://github.com/hashicorp/consul/blob/main/proto/pbcommon/common.go#L148-L152

And I don't see a field on `pbcommon.ReadRequest` to set AllowStale.

Reply:
Oops, you're right, we have the plumbing supported for this in `ForwardGRPC` and `canServeReadRequest`, but I didn't realize `pbcommon.ReadRequest` was never updated for it. Sorry for the confusion @kisunji!